[REF] partner_firstname,partner_second_lastname: Avoid _inverse_name when recalculating names

payen000 opened the pull request:
Currently, pressing the Recalculate Names button in the 'Partner Names Order' setting
triggers the _inverse_name method which can modify the firstname, lastname and lastname2
of a partner depending on the 'partner_names_order' setting and the number of names
in each field.

Example video: https://www.youtube.com/watch?v=vZ5FeMoTqRU

The modification to action_recalculate_partners_name prevents the _inverse_name method from 
triggering when pressing Recalculate Names, leaving the firstname, lastname and lastname2 
fields intact.

Example after applying changes: https://youtu.be/8WRSU2G86O0

Additionally, added related unittest to ensure names are computed correctly without
modifying firstname, lastname or lastname2.

payen000 committed:
[REF] partner_firstname,partner_second_lastname: Avoid _inverse_name when recalculating names

The modification to action_recalculate_partners_name prevents the _inverse_name method from
triggering when recalculating the names of partners, leaving the firstname, lastname and
lastname2 fields intact.

hugho-ad commented on partner_second_lastname/models/res_config_settings.py:
```diff
@@ -33,3 +36,18 @@ def _partners_for_recalculating(self):
         )
+
+    def action_recalculate_partners_name(self):
+        # Override of partner_firstname's action_recalculate_partners_name
```
move this to a docstring

```python
def action_recalculate_partners_name(self):
    """DOC"""

hugho-ad commented on partner_second_lastname/models/res_config_settings.py:
```diff
@@ -33,3 +36,18 @@ def _partners_for_recalculating(self):
+        )
+        partners = self._partners_for_recalculating()
+        _logger.info("Recalculating names for %d partners.", len(partners))
+        self.env.add_to_compute(self.env["res.partner"]._fields["name"], partners)
```
this is the main change right? 

why not doing it in original method definition?

payen000 replied:
I was trying to keep this change and the unittest in the same module, but since partner_second_lastname depends on partner_firstname, there will be no dependency problems, so there's no reason to override after all, thank you!

payen000 commented:
Hello again @hugho-ad, I've applied your suggestions, would you review again, please? :)

luisg123v commented on partner_second_lastname/tests/__init__.py:
```diff
@@ -2,4 +2,5 @@
 
 from . import test_name
 from . import test_config
+from . import test_recalculate_names
```
This could be included into `test_name` IMHO

luisg123v commented on partner_firstname/models/base_config_settings.py:
```diff
@@ -62,7 +62,7 @@ def action_recalculate_partners_name(self):
         partners = self._partners_for_recalculating()
         _logger.info("Recalculating names for %d partners.", len(partners))
-        partners._compute_name()
+        self.env.add_to_compute(self.env["res.partner"]._fields["name"], partners)
```
A comment explaining why computation is delayed instead of manually calling compute method would be great.

luisg123v commented:
This should be into separated commits I think. When migrated, `partner_firstname` is usually migrated before `partner_second_lastname`. So when they will be migrated, this commit will be duplicated.

payen000 committed:
[REF] partner_second_lastname: Added unittest related to name recalculation

This unittest was added to ensure that, after pressing on 'Recalculate Names'
button in 'partner_names_order' setting:

1 - firstname, lastname and lastname2 fields are not modified, and
2 - name field is computed correctly.

moylop260 commented:
/ocabot merge minor

moylop260 commented:
/ocabot merge minor
